File: graph/unweighted_graph/removeEdge.php
<?php 

class RemoveEdge{
    private $adjacency_list = [];
    

    public function addVertex($vertex){
        if(!array_key_exists($vertex,$this->adjacency_list)){
            $this->adjacency_list[$vertex]  = [];
        }
    }

    public function addEdge($vertex1,$vertex2){
        $this->addVertex($vertex1);
        $this->addVertex($vertex2);

        $this->adjacency_list[$vertex1][] = $vertex2;
        $this->adjacency_list[$vertex2][] = $vertex1;
    }

    public function removeEdge($vertex1, $vertex2){
        if (isset($this->adjacency_list[$vertex1])) {
            $this->adjacency_list[$vertex1] = array_values(array_filter(
                $this->adjacency_list[$vertex1],
                function ($v) use ($vertex2) {
                    return $v !== $vertex2;
                }
            ));
        }

        if (isset($this->adjacency_list[$vertex2])) {
            $this->adjacency_list[$vertex2] = array_values(array_filter(
                $this->adjacency_list[$vertex2],
                function ($v) use ($vertex1) {
                    return $v !== $vertex1;
                }
            ));
        }
    }

    public function removeVertex($vertex) {
        if (isset($this->adjacency_list[$vertex])) {
            foreach ($this->adjacency_list[$vertex] as $neighbor) {
                $this->removeEdge($vertex, $neighbor);
            }
            unset($this->adjacency_list[$vertex]);
        }
    }


    public function printGraph(){
        foreach($this->adjacency_list as $vertex => $neighbors){
            echo "$vertex -> ". implode(', ', $neighbors) . "\n";
        }
    }
}

$graph = new RemoveEdge();
$graph->addEdge("A", "B");
$graph->addEdge("A", "C");
$graph->addEdge("B", "D");
$graph->addEdge("C", "D");
$graph->addEdge("C", "E");

echo "Adjacency List:" . PHP_EOL;
$graph->printGraph();

$graph->removeEdge("A", "C");

echo PHP_EOL . "After Removing Edge:" . PHP_EOL;
$graph->printGraph();

File: grap/remove_edge.py
class Graph:
    def __init__(self):
        self.adj_list = {}

    def print_graph(self):
        for vertex in self.adj_list:
            print(vertex, ':', self.adj_list[vertex])
    
    def add_vertex(self, vertex):
        if vertex not in self.adj_list.keys():
            self.adj_list[vertex] = []
            return True
        return False
        
    def add_edge(self, v1, v2):
        if v1 in self.adj_list.keys() and v2 in self.adj_list.keys():
            self.adj_list[v1].append(v2)
            self.adj_list[v2].append(v1)
            return True
        return False
    
    def remove_edge(self, v1, v2):
        if v1 in self.adj_list.keys() and v2 in self.adj_list.keys():
            try:
                self.adj_list[v1].remove(v2) 
                self.adj_list[v2].remove(v1)
            except ValueError:
                pass
            return True 
        return False

    
graph = Graph()
graph.add_vertex("A")
graph.add_vertex('B')
graph.add_vertex('C')
graph.add_vertex('D')

graph.add_edge('A','B')
graph.add_edge('B', 'C')
graph.add_edge('C', 'A')

graph.remove_edge('A','B')
graph.remove_edge('A','D')

graph.print_graph()